src/renderer/components/dialogs/Settings.js

```diff
@@ -93,26 +103,37 @@ class Settings extends React.Component {
     this.props.userFeedback({ type: 'success', text: 'Account updated successfully' })
   }
 
+  onAdvancedSettingsOpen () {
+    ipcRenderer.send('retrieveAdvancedSettingsObj')
+  }
+
+  retrieveAdvancedSettingsObjResp (e, advancedSettingsObj) {
+    console.log('retrieveAdvancedSettingsObjResp', advancedSettingsObj)
+    this.setState({ advancedSettings: advancedSettingsObj })
+  }
+
   render () {
     const { deltachat, isOpen, onClose } = this.props
     const { advancedSettings, saved, keyTransfer } = this.state
 
+    console.log(deltachat.credentials)
+
     const tx = window.translate
     const title = tx('settingsTitle')
 
     return (
       <div>
         <KeyTransfer isOpen={keyTransfer} onClose={this.onKeyTransferComplete} />
         <Dialog
-          isOpen={advancedSettings}
+          isOpen={advancedSettings !== false}
           title={tx('settingsAccountTitle')}
           icon='settings'
           onClose={() => this.setState({ advancedSettings: false })}>
           <SettingsDialog className={Classes.DIALOG_BODY}>
             <Card elevation={Elevation.ONE}>
               <H5>{tx('settingsAccountTitle')}</H5>
               <Login
-                addr={deltachat.credentials.addr}
+                {...advancedSettings}
```

Comment:
I still don't get what the result of this is.

Reply:
this expands the object into it's parts

Reply:
So if `var obj = { foo: true, bar: false }` then `<Component {...obj} />` would be the same as `<Component obj={true} bar={false} />` ?

Reply:
Think I got it now. It would be the same as `<Component foo={obj.foo} bar={obj.bar} /> `

Reply:
Yup would be the same :)